perun-oidc-server/src/main/java/cz/muni/ics/oauth2/web/OAuthConfirmationController.java

```diff
@@ -62,20 +69,54 @@
 @Slf4j
 public class OAuthConfirmationController {
 
-	@Autowired
-	private ClientDetailsEntityService clientService;
+	public static final String AUTHORIZATION_REQUEST = "authorizationRequest";
+	public static final String ERROR = "error";
+	public static final String INTERACTION_REQUIRED = "interaction_required";
+	public static final String STATE = "state";
+	public static final String NONE = "none";
+	public static final String REDIRECT = "redirect";
+	public static final String CODE = "code";
+	public static final String AUTH_REQUEST = "auth_request";
+	public static final String CLIENT = "client";
+	public static final String REDIRECT_URI = "redirect_uri";
+	public static final String SCOPES = "scopes";
+	public static final String CLAIMS = "claims";
+	public static final String CONTACTS = "contacts";
+	public static final String GRAS = "gras";
+	public static final String DEFAULT = "default";
+	public static final String PAGE = "page";
+	public static final String CONSENT = "consent";
+	public static final String THEMED_APPROVE = "themedApprove";
+	public static final String APPROVE = "approve";
 
-	@Autowired
+	private ClientDetailsEntityService clientService;
 	private SystemScopeService scopeService;
-
-	@Autowired
 	private ScopeClaimTranslationService scopeClaimTranslationService;
-
-	@Autowired
 	private UserInfoService userInfoService;
+	private RedirectResolver redirectResolver;
+	private PerunOidcConfig perunOidcConfig;
+	private WebHtmlClasses htmlClasses;
+	private PerunScopeClaimTranslationService perunScopeClaimTranslationService;
 
 	@Autowired
-	private RedirectResolver redirectResolver;
+	public OAuthConfirmationController(ClientDetailsEntityService clientService,
+									   SystemScopeService scopeService,
+									   ScopeClaimTranslationService scopeClaimTranslationService,
+									   UserInfoService userInfoService,
+									   RedirectResolver redirectResolver,
+									   PerunOidcConfig perunOidcConfig,
+									   WebHtmlClasses htmlClasses,
+									   PerunScopeClaimTranslationService perunScopeClaimTranslationService) {
+
+		this.clientService = clientService;
+		this.scopeService = scopeService;
+		this.scopeClaimTranslationService = scopeClaimTranslationService;
```

Comment:
This bean, and the bean on line 118 are actually the same one. Just pass the more generic one if possible. If it needs the PerunScope... type, pass that one instead.